.github/workflows/doc.yml

```diff
@@ -0,0 +1,32 @@
+# Based on https://gist.github.com/domenic/ec8b0fc8ab45f39403dd
+name: Build Docs
+on:
+  pull_request:
+    branches:
+    - master
+  push:
+    branches:
+    - master
```

Comment:
Change this to:
```yml
on:
  push:
    tags:
    - 'doc-v*'
```

... (or something similar) and documentation can be versioned by creating tags prefixed with `doc-v`:
```sh
$ git tag doc-v1.4.6
```

This way, docs will only be rebuilt on `gh-pages` branch when one of these tags is pushed.